File: src/main/java/com/kamikazejam/datastore/base/log/PluginLogger.java
package com.kamikazejam.datastore.base.log;

import com.kamikazejam.datastore.DataStoreSource;
import org.bukkit.plugin.Plugin;

public class PluginLogger extends LoggerService {
    public PluginLogger() {}

    @Override
    public String getLoggerName() {
        return "DataStore";
    }

    @Override
    public Plugin getPlugin() {
        return DataStoreSource.get();
    }

    @Override
    public boolean isDebug() {
        return DataStoreSource.isDebug();
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/base/log/PluginLogger.kt
package com.kamikazejam.datastore.base.log

import com.kamikazejam.datastore.DataStoreSource
import org.bukkit.plugin.Plugin

class PluginLogger : LoggerService() {
    override val loggerName: String
        get() = "DataStore"

    override val plugin: Plugin
        get() = DataStoreSource.get()

    override val isDebug: Boolean
        get() = DataStoreSource.isDebug()
}
